src/persistReducers.js

```diff
@@ -53,117 +55,106 @@ export default function persistReducer(config, baseReducer) {
     let _persist = state && state.has('_persist') && state.get('_persist').size && state.get('_persist') || null;
     let restState = state && state.delete('_persist');
 
-    const handlesMap = {
-      [PERSIST]: function () {
-        let _sealed = false
-        let _rehydrate = (payload, err) => {
-          // dev warning if we are already sealed
-          if (process.env.NODE_ENV !== 'production' && _sealed)
-            console.error(
-              `redux-persist: rehydrate for "${
-              config.key
-              }" called after timeout.`,
-              payload,
-              err
-            )
-
-          // only rehydrate if we are not already sealed
-          if (!_sealed) {
-            action.rehydrate(config.key, payload, err)
-            _sealed = true
+      if (action.type === PERSIST) {
```

Comment:
I restored 'if else' template action handling due to issue with multiple nested reducer. 'if else' handling provide same functionality as earlier handle[action] but do not require 'return' on hydration while config key and action key do not equal.